core/src/test/java/org/elasticsearch/index/mapper/RangeFieldMapperTests.java

```diff
@@ -0,0 +1,325 @@
+
+        mapping = XContentFactory.jsonBuilder().startObject().startObject("type")
+            .startObject("properties").startObject("field").field("type", type);
+        if (type.equals("date_range")) {
+            mapping = mapping.field("format", DATE_FORMAT);
+        }
+        mapping = mapping.field("include_in_all", false).endObject().endObject()
+            .endObject().endObject();
+
+        mapper = parser.parse("type", new CompressedXContent(mapping.string()));
+
+        assertEquals(mapping.string(), mapper.mappingSource().toString());
+
+        doc = mapper.parse("test", "type", "1", XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("field")
+            .field(getFromField(), getFrom(type))
+            .field(getToField(), getTo(type))
+            .endObject()
+            .endObject().bytes());
+
+        fields = doc.rootDoc().getFields("_all");
+        assertEquals(0, fields.length);
+    }
+
+    @Override
+    protected void doTestNullValue(String type) throws IOException {
+        XContentBuilder mapping = XContentFactory.jsonBuilder().startObject().startObject("type")
+            .startObject("properties").startObject("field").field("type", type).field("store", true);
+        if (type.equals("date_range")) {
+            mapping = mapping.field("format", DATE_FORMAT);
+        }
+        mapping = mapping.endObject().endObject().endObject().endObject();
+
+        DocumentMapper mapper = parser.parse("type", new CompressedXContent(mapping.string()));
+        assertEquals(mapping.string(), mapper.mappingSource().toString());
+
+        ParsedDocument doc = mapper.parse("test", "type", "1", XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("field")
+            .field(getFromField(), getFrom(type))
+            .nullField(getToField())
+            .endObject()
+            .endObject().bytes());
+        assertEquals(2, doc.rootDoc().getFields("field").length);
+
+        doc = mapper.parse("test", "type", "1", XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("field")
+            .field(getFromField(), getFrom(type))
+            .nullField(getToField())
+            .endObject()
+            .endObject().bytes());
+
+        IndexableField[] fields = doc.rootDoc().getFields("field");
+        assertEquals(2, fields.length);
+        IndexableField pointField = fields[0];
+        assertEquals(2, pointField.fieldType().pointDimensionCount());
+        assertFalse(pointField.fieldType().stored());
+        IndexableField storedField = fields[1];
+        assertTrue(storedField.fieldType().stored());
+        assertThat(storedField.stringValue(), containsString(type.equals("date_range") ? "1477872000000" : "5"));
+        assertThat(storedField.stringValue(), containsString(getMax(type) + ""));
+    }
```

Comment:
I'd also like a test for the case that either a single bound or none of the bounds are specified (even if that means checking that an exception is thrown depending on the decision we make).